Description: use libc::write directly for validation.
 Calling nix::unistd::write requires a slice, which we cannot soundly
 construct from a potentilly invalid pointer. This causes panics
 in debug mode with the current version of rustc.

 To avoid this we call libc::write directly.
Author: Peter Michael Green <plugwash@debian.org>

--- rust-pprof-0.13.0.orig/src/addr_validate.rs
+++ rust-pprof-0.13.0/src/addr_validate.rs
@@ -5,8 +5,9 @@ use std::{
 
 use nix::{
     errno::Errno,
-    unistd::{close, read, write},
+    unistd::{close, read},
 };
+use libc::write;
 
 struct Pipes {
     read_fd: AtomicI32,
@@ -90,9 +91,7 @@ pub fn validate(addr: *const libc::c_voi
 
     let write_fd = MEM_VALIDATE_PIPE.write_fd.load(Ordering::SeqCst);
     loop {
-        let buf = unsafe { std::slice::from_raw_parts(addr as *const u8, CHECK_LENGTH) };
-
-        match write(write_fd, buf) {
+        match Errno::result( unsafe { write(write_fd, addr, CHECK_LENGTH) } ) {
             Ok(bytes) => break bytes > 0,
             Err(_err @ Errno::EINTR) => continue,
             Err(_) => break false,
